src/Framework/TaskItemData.cs

```diff
@@ -25,6 +25,26 @@ public TaskItemData(string itemSpec, IDictionary<string, string> metadata)
             Metadata = metadata ?? _emptyMetadata;
         }
 
+        /// <summary>
+        /// Clone the task item and all metadata to create a snapshot
+        /// </summary>
+        /// <param name="original">An <see cref="ITaskItem"/> to clone</param>
+        public TaskItemData(ITaskItem original)
+        {
+            ItemSpec = original.ItemSpec;
+            var metadata = original.EnumerateMetadata();
```

Comment:
Isn't there a CopyMetadataTo function? There also seems to be CloneCustomMetadata.

Reply:
I wasn't sure if CloneCustomMetadata will work across appdomains so decided not to risk it.